arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/popup/views/PopupView.kt

```diff
@@ -0,0 +1,331 @@
+                if (isEditMode) {
+                    val codedValueDomain: CodedValueDomain? =
+                        popupManager.getDomain(popupField) as? CodedValueDomain
+                    if (codedValueDomain != null) {
+                        setUpSpinner(codedValueDomain, popupField)
+                        valueEditText.visibility = View.GONE
+                        valueTextView.visibility = View.GONE
+                        codedValueDomainSpinner.visibility = View.VISIBLE
+                        separatingLineView.visibility = View.VISIBLE
+                    } else {
+                        valueEditText.inputType =
+                            getInputType(popupManager.getFieldType(popupField))
+                        valueEditText.visibility = View.VISIBLE
+                        separatingLineView.visibility = View.GONE
+                        valueTextView.visibility = View.GONE
+                        //save original colors
+                        val oldColors: ColorStateList = labelTextView.textColors
+                        // here we assign and hold the values of the editable fields, entered by the user
+                        // in popupAttribute.tempValue
+                        valueEditText.doAfterTextChanged {
+                            if (valueEditText.hasFocus()) {
+
+                                val validationError: ArcGISRuntimeException? = updateValue(
+                                    popupField,
+                                    valueEditText.text.toString()
+                                )
+                                if (validationError != null) {
+                                    val fieldLabelWithValidationError =
+                                        popupField.label + ": " + validationError.message
+                                    labelTextView.text = fieldLabelWithValidationError
+                                    labelTextView.setTextColor(Color.RED)
+                                } else {
+                                    labelTextView.text = popupField.label
+                                    labelTextView.setTextColor(oldColors)
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    valueEditText.visibility = View.GONE
+                    codedValueDomainSpinner.visibility = View.GONE
+                    valueTextView.visibility = View.VISIBLE
+                    separatingLineView.visibility = View.VISIBLE
+                }
+            }
+        }
+
+        /**
+         * Sets up spinner for PopupFields that have a CodedValueDomain.
+         */
+        private fun setUpSpinner(codedValueDomain: CodedValueDomain, popupField: PopupField) {
+            val codedValuesNames = mutableListOf<String>()
+            codedValueDomain.codedValues.forEach { codedValue -> codedValuesNames.add(codedValue.name) }
```

Comment:
I think specifying the argument name here makes the code more readable, rather than using the implicit default name `it` . Happy to change it if we have decided as team to use the implicit name all the time.